addons/mail/static/src/discuss/core/web/discuss_sidebar_categories.js

```diff
@@ -84,7 +84,9 @@ export class DiscussSidebarCategories extends Component {
     filteredThreads(category) {
         return category.threads.filter((thread) => {
             return (
-                (thread.is_pinned || thread.group_based_subscription) &&
+                (thread.is_pinned ||
+                    thread.eq(this.store.discuss.thread) ||
+                    thread.group_based_subscription) &&
```

Comment:
not related to PR but condition `thread.group_based_subscription` is not really correct, it should be `channel_type == "channel" and is member` instead